main/admin/user_advanced_edit.php

```diff
@@ -0,0 +1,330 @@
+            let fieldName = $(this).attr("name").replace("[" + userId + "]", "");
+            userData[fieldName] = $(this).val();
+        });
+
+        $("input[type=\'radio\'][name$=\'[" + userId + "]\']:checked").each(function() {
+            let fieldName = $(this).attr("name").replace("[" + userId + "]", "");
+            userData[fieldName] = $(this).val();
+        });
+
+        $("input[type=\'checkbox\'][name$=\'[" + userId + "]\']:checked").each(function() {
+            let fieldName = $(this).attr("name").replace("[" + userId + "]", "");
+            userData[fieldName] = "1";
+        });
+
+        $("input[name^=\'extra_[" + userId + "]\'], select[name^=\'extra_[" + userId + "]\'], textarea[name^=\'extra_[" + userId + "]\']").each(function() {
+            let fieldName = $(this).attr("name").replace("extra_[" + userId + "]", "extra_");
+
+            if ($(this).hasClass("tags-input")) {
+                userData[fieldName] = $(this).val().split(",");
+            }
+            else if ($(this).hasClass("doubleselect-input")) {
+                let values = $(this).val().split(",");
+                if (values.length === 2) {
+                    userData[fieldName] = values[0];
+                    userData[fieldName + "_second"] = values[1];
+                }
+            }
+            else {
+                userData[fieldName] = $(this).val();
+            }
+        });
+
+        return userData;
+    }
+
+    $(".saveUser").click(function() {
+        let userId = $(this).data("user-id");
+        if (!userId) {
+            return;
+        }
+
+        let userData = getUserData(userId);
+
+        $.post("'.api_get_path(WEB_AJAX_PATH).'user_manager.ajax.php", {
+            a: "update_users",
+            users: JSON.stringify([userData])
+        }, function(response) {
+            alert(response.message);
+        }, "json");
+    });
+
+    $("#saveAll").click(function() {
+        let usersData = [];
+
+        $(".saveUser").each(function() {
+            let userId = $(this).data("user-id");
+            let userData = getUserData(userId);
+            if (userData) usersData.push(userData);
+        });
+
+        if (usersData.length === 0) {
+            return;
+        }
+
+        $.post("'.api_get_path(WEB_AJAX_PATH).'user_manager.ajax.php", {
+            a: "update_users",
+            users: JSON.stringify(usersData)
+        }, function(response) {
+            alert(response.message);
+        }, "json");
+    });
+
+});
+</script>';
+
+$formContent = $form->returnForm();
+
+// Render page
+$tpl = new Template($tool_name);
+$tpl->assign('actions', $toolbarActions);
+$tpl->assign('message', $message);
+$tpl->assign('content', $formContent . $fieldSelector . $tableResult . (!empty($users) ? '<button class="btn btn-success" id="saveAll">'.get_lang('SaveAll').'</button>' : ''));
```

Comment:
"SaveAll" could be replace (in get_lang()) by SaveChanges, to avoid introducing a new variable.